pennylane/templates/subroutines/qrom.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the template for QROM.
+"""
+
+import math
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import Operation
+
+
+def _multi_swap(wires1, wires2):
+    """Apply a series of SWAP gates between two sets of wires."""
+    for wire1, wire2 in zip(wires1, wires2):
+        qml.SWAP(wires=[wire1, wire2])
```

Comment:
Just fyi, 

If the set of wires don't have the same size, then `zip(wires1, wires2)` will zip together the first `n` wires where `n = min(len(wires1), len(wires2))`. Probably you are only using this on wires with the same length. Just want to let you know if you think we should check if they are the same length or not.

Reply:
As defined, they will always have the same dimension, but thanks for pointing it out!